docs/library/forms/form.md

```diff
@@ -0,0 +1,527 @@
+
+```python demo
+rdxp.form_root(
+    rdxp.form_field(
+        rdxt.flex(
+            rdxp.form_label("Email"),
+            rdxp.form_control(
+                rdxt.textfield_input(
+                    placeholder="Email Address",
+                    # type attribute is required for "typeMismatch" validation
+                    type="email",
+                ),
+                as_child=True,
+            ),
+            rdxp.form_message("Please enter a valid email", match="typeMismatch"),
+            rdxp.form_submit(
+                  rdxt.button("Submit"),
+                  as_child=True,
+            ),
+            direction="column",
+            gap="2",
+            align="stretch",
+        ),
+        name="email",
+    ),
+    on_submit=lambda _: rx.window_alert("Submitted!"),
+    reset_on_submit=True,
+)
+```
+
+In this example, the `textfield_input` has an attribute `type="email"` and the `form_message` attribute `match="typeMismatch"`. Those are required for the form to validate the input by its type. The prop `as_child="True"` is required when we use other components to construct `form_control` or `form_submit`. In the example, we have used `textfield_input` to construct the `form_control` and `button` the `form_submit`.
+
+## Form Anatomy
+
+```python eval
+rx.code_block(
+    """form_root(
+    form_field(
+        form_label(...),
+        form_control(...),
+        form_message(...),
+    ),
+    form_submit(...),
+)""",
+    language="python",
+)
+```
+
+A `form_root` contains all the parts of a form. The `form_field`, `form_submit`, etc should all be inside a `form_root`. A `form_field` can contain a `form_label`, a `form_control`, and a `form_message`. A `form_label` is a label element. A `form_control` is where the user enters the input or makes selections. By default, `form_control` is a input. You can use your own form components to construct the `form_control` and set the prop `as_child=True`.
+
+```md alert info
+The current version of Radix Forms does not support composing `form_control` with other Radix form primitives such as `checkbox`, `select`, etc.
+```
+
+The `form_message` is a validation message which is automatically wired (functionality and accessibility). When `form_control` determines the input is invalid, the `form_message` is shown. You set the `match` prop to enable [client side validation](#client-side-validation). You can set the `force_match` prop of `form_message` and `server_invalid` prop of `form_field` together to perform [server side validation](#server-side-validation).
```

Comment:
This whole section feels very dense and hard to read. It may be okay for now, I think @masenf is going through for voice, but we should simplify this

Reply:
is it the excessive use of `code`? I rewrote some of the content by using the proper nouns instead of the code. Also did a first pass using the documentation voice tips from Masen. I feel some of the stuff probably requires a short discussion: the tips are somewhat still up to interpretation. We can do another pass when we scrub the docs